dms-service/src/main/java/de/muenchen/oss/swim/dms/adapter/out/dms/DmsAdapter.java

```diff
@@ -0,0 +1,82 @@
+import de.muenchen.refarch.integration.dms.model.CreateIncomingAntwortDTO;
+import de.muenchen.refarch.integration.dms.model.CreateIncomingBasisAnfrageDTO;
+import de.muenchen.refarch.integration.dms.model.CreateObjectAndImportToInboxDTO;
+import de.muenchen.oss.swim.dms.application.port.out.DmsOutPort;
+import de.muenchen.oss.swim.dms.domain.exception.DmsException;
+import de.muenchen.oss.swim.dms.domain.model.DmsTarget;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.io.AbstractResource;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class DmsAdapter implements DmsOutPort {
+    private final ObjectAndImportToInboxApi objectAndImportToInboxApi;
+    private final IncomingsApi incomingsApi;
+
+    private final static String DMS_APPLICATION = "SWIM";
+
+    @Override
+    public void putFileInInbox(final DmsTarget dmsTarget, final String fileName, final InputStream inputStream) {
+        log.debug("Putting file {} in inbox {}", fileName, dmsTarget);
+        final CreateObjectAndImportToInboxDTO request = new CreateObjectAndImportToInboxDTO();
+        request.setObjaddress(dmsTarget.coo());
+        try {
+            // FIXME directly use InputStream
+            final AbstractResource file = new NamedByteArrayRessource(fileName, inputStream.readAllBytes());
+            objectAndImportToInboxApi.createObjectAndImportToInbox(
+                    request,
+                    DMS_APPLICATION,
+                    dmsTarget.userName(),
+                    null,
+                    null,
+                    List.of(file)).block();
+            log.info("Created new Object {} for Inbox {}", fileName, dmsTarget);
+        } catch (final IOException e) {
+            throw new DmsException("Error while handling input stream", e);
+        } catch (final WebClientResponseException e) {
+            throw new DmsException(String.format("Dms request failed with message: %s", e.getResponseBodyAsString()), e);
+        }
+    }
+
+    @Override
+    public String createIncoming(final DmsTarget dmsTarget, final String incomingName, final String contentObjectName, final InputStream inputStream) {
+        log.debug("Putting file {} in procedure {}", contentObjectName, dmsTarget);
+        final CreateIncomingBasisAnfrageDTO request = new CreateIncomingBasisAnfrageDTO();
+        request.referrednumber(dmsTarget.coo());
+        request.shortname(incomingName);
+        request.useou(true);
+        try {
+            // FIXME directly use InputStream
+            final AbstractResource file = new NamedByteArrayRessource(contentObjectName, inputStream.readAllBytes());
+            final CreateIncomingAntwortDTO response = incomingsApi.eingangZuVorgangAnlegen(
```

Comment:
das "eingangZuVorgangAnlegen" finde ich in der Swagger-GUI gar nicht, nur in der api-doc als "operationId". Was ist das denn und wo kommt das her? Müsste das von der Logik her nicht einfach "createIncomings" heißen?

Reply:
Bin da mit den Standard-Einstellungen für den openapi-generator gegangen.
`createIncoming` gibt es so halt nirgendwo, denke an sich hat er schon recht, das so zu machen, ist jetzt nur durch den Deutsch Englisch Mix etwas seltsam.

Reply:
Ja, das kommt wohl aus der api-doc. Ich habe mal bei den Kollegen angefragt, warum sie das an dieser Stelle anders benannt haben als bspw. bei CreateObjectAndImportToInbox.